src/blockchain/node/wallet/subchain/statemachine/Scan.cpp

```diff
@@ -277,7 +288,16 @@ auto Scan::Imp::work() noexcept -> bool
         }());
     }
 
-    return (false == caught_up());
+    return !caught_up() ? SM_Scan_fast : SM_Scan_slow;
+}
+
+network::zeromq::Message Scan::Imp::make_work(
+    Vector<ScanStatus>&& vec) const noexcept
+{
+    auto work = MakeWork(Work::update);
+    add_last_reorg(work);
+    encode(vec, work);
```

Comment:
should not we move vector into encode ?

Reply:
Scan.cpp defines and calls the same lambda several times in a context, where the lambda should really be a private member function. I believe this was corrected at one point, but then some poor misguided soul restored the lambdas in the branch develop and automatic merge has left the private function intact. Other than it is no longer used, the function is correct. I'm about to remove it leaving the restored lambdas for now, but at some stage, when we try to squeeze all this onto a mobile platform, the issue should be revisited.